### PR TITLE
feat: qr code generation

### PR DESCRIPTION
**Pull request by yehjxraymond**

### CODE REVIEW

**Comment by john-dot-oa** [comment]
## Application
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=36

## Storybook
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=storybook-36

**Comment by john-dot-oa** [comment]
## Application
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=36

## Storybook
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=storybook-36

**Comment by john-dot-oa** [comment]
## Application
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=36

## Storybook
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=storybook-36

**Comment by john-dot-oa** [comment]
## Application
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=36

## Storybook
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=storybook-36

File: src/components/Layout/BottomSheet.tsx [hunk 1]
```diff
@@ -34,7 +43,8 @@ export const BottomSheet: FunctionComponent<BottomSheet> = ({
   // this allows snapPoints to be dynamic
   useEffect(() => {
     bottomSheetRef.current && bottomSheetRef.current.snapTo(0);
-  }, [snapPoints]);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [...snapPoints]); //
```
Comment (sebastianquek) [inline src/components/Layout/BottomSheet.tsx]:
```suggestion
  }, [...snapPoints]);
```

File: src/components/DocumentRenderer/QrCode/QrCode.test.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import React from "react";
+import { render } from "@testing-library/react-native";
+import { QrCode } from "./index";
+
+describe("QrCode", () => {
+  it("should not display anything if both qrCode and qrCodeLoading is false", () => {
+    expect.assertions(1);
+    const { asJSON } = render(<QrCode />);
+    expect(asJSON().children).toHaveLength(0);
```
Comment (sebastianquek) [inline src/components/DocumentRenderer/QrCode/QrCode.test.tsx]: `render` provides a `container` property, so we don't need to convert to JSON.
```
    const { container } = render(<QrCode />);
    expect(container.children).toHaveLength(0);
```

File: src/components/DocumentRenderer/LocalDocumentRendererContainer.tsx [hunk 3]
```diff
@@ -27,7 +29,12 @@ export const LocalDocumentRendererContainer: FunctionComponent<NavigationProps>
         document={document.document}
         goBack={() => navigation.goBack()}
       />
-      <DocumentDetailsSheet document={document.document} />
+      <DocumentDetailsSheet
+        document={document.document}
+        onGenerateQr={generateQr(document.document)}
+        qrCode={qrCode}
+        qrCodeLoading={qrCodeLoading}
```
Comment (sebastianquek) [inline src/components/DocumentRenderer/LocalDocumentRendererContainer.tsx]: Hmm, given that all the qr-related props are being passed down, should we let `DocumentDetailsSheet` handle the qr related stuff instead?

**Comment by john-dot-oa** [comment]
## Application
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=36

## Storybook
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=storybook-36

**Comment by john-dot-oa** [comment]
## Application
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=36

## Storybook
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=storybook-36

**Comment by john-dot-oa** [comment]
## Application
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=36

## Storybook
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=storybook-36

**Comment by john-dot-oa** [comment]
## Application
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=36

## Storybook
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=storybook-36

**Pull request merged**